### PR TITLE
CapturePoint indicator HUD

### PR DESCRIPTION
**Pull request by Tleety**
## Added ##

**CapturePointArrowHUD** components, add this to an object that should rotate towards capture points, this needs a team component to know what capturepoint it should track.
**CapturePointArrowHUDSystem**, this track every arrow in play and rotates it towards the next capture point
**Arrow** added to blue players hud.

### CODE REVIEW

File: src/Game/Systems/CapturePointArrowHUDSystem.cpp [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    : System(params)
+    , ImpureSystem()
+{
+    EVENT_SUBSCRIBE_MEMBER(m_ECapturedEvent, &CapturePointArrowHUDSystem::OnCapturePointCaptured);
+}
+
+
+void CapturePointArrowHUDSystem::Update(double dt)
+{
+    bool LoadCheck = true;
```
Comment (JockeBTH) [inline src/Game/Systems/CapturePointArrowHUDSystem.cpp]: Please have a look at the code style guide. LoadCheck, ArrowHUDs, CapturePoints etc. does not follow it. :)

File: src/Game/Systems/CapturePointArrowHUDSystem.cpp [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        if (!ArrowEntity.Valid()) {
+            continue;
+        }
+        if(!ArrowEntity.HasComponent("Team")) {
+            continue;
+        }
+        auto cTeam = ArrowEntity["Team"];
+        int currentTeam = (int)cTeam["Team"];
+
+        if (LoadCheck) {
```
Comment (JockeBTH) [inline src/Game/Systems/CapturePointArrowHUDSystem.cpp]: If a player connects after one or more capture points have been taken, what will happen?

File: src/Game/Systems/CapturePointArrowHUDSystem.cpp [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        if (parent.Valid()) {
+            arrowOri -= Transform::AbsoluteOrientationEuler(parent);
+        }
+    }
+}
+
+bool CapturePointArrowHUDSystem::OnCapturePointCaptured(Events::Captured& e)
+{
+    if (!e.NextCapturePoint.HasComponent("Team"))
+    {
```
Comment (Stiffly) [inline src/Game/Systems/CapturePointArrowHUDSystem.cpp]: Code Rules

File: src/Game/Systems/CapturePointArrowHUDSystem.cpp [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+    auto cTeam = e.NextCapturePoint["Team"];
+    
+    int redTeam = (int)cTeam["Team"].Enum("Red");
+    int blueTeam = (int)cTeam["Team"].Enum("Blue");
+    int spectatorTeam = (int)cTeam["Team"].Enum("Spectator");
+    int target = -1;
+
+    if (e.NextCapturePoint.HasComponent("CapturePoint")) {
+        target = (int)e.NextCapturePoint["CapturePoint"]["CapturePointNumber"];
```
Comment (JockeBTH) [inline src/Game/Systems/CapturePointArrowHUDSystem.cpp]: Not used variable target?

File: src/Game/Systems/CapturePointArrowHUDSystem.cpp [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+    } else {
+        return 0;
+    }
+
+    if(e.TeamNumberThatCapturedCapturePoint == redTeam) {
```
Comment (JockeBTH) [inline src/Game/Systems/CapturePointArrowHUDSystem.cpp]: What will happen when an enemy takes your capture point? Does the target change to a previousCapturePoint?

**Comment by JockeBTH** [comment]
Good job! :thumbsup:

**Pull request merged**